proxy/http/remap/RemapConfig.h

```diff
@@ -23,58 +23,171 @@
 
 #pragma once
 
+#include <list>
 #include "AclFiltering.h"
+#include "tscore/ts_file.h"
+#include "UrlRewrite.h"
 
 class UrlRewrite;
-
-#define BUILD_TABLE_MAX_ARGS 2048
-
-// Remap inline options
-#define REMAP_OPTFLG_MAP_WITH_REFERER 0x0001u /* "map_with_referer" option */
-#define REMAP_OPTFLG_PLUGIN 0x0002u           /* "plugin=" option (per remap plugin) */
-#define REMAP_OPTFLG_PPARAM 0x0004u           /* "pparam=" option (per remap plugin option) */
-#define REMAP_OPTFLG_METHOD 0x0008u           /* "method=" option (used for ACL filtering) */
-#define REMAP_OPTFLG_SRC_IP 0x0010u           /* "src_ip=" option (used for ACL filtering) */
-#define REMAP_OPTFLG_ACTION 0x0020u           /* "action=" option (used for ACL filtering) */
-#define REMAP_OPTFLG_INTERNAL 0x0040u         /* only allow internal requests to hit this remap */
-#define REMAP_OPTFLG_IN_IP 0x0080u            /* "in_ip=" option (used for ACL filtering)*/
-#define REMAP_OPTFLG_MAP_ID 0x0800u           /* associate a map ID with this rule */
-#define REMAP_OPTFLG_INVERT 0x80000000u       /* "invert" the rule (for src_ip at least) */
-#define REMAP_OPTFLG_ALL_FILTERS (REMAP_OPTFLG_METHOD | REMAP_OPTFLG_SRC_IP | REMAP_OPTFLG_ACTION | REMAP_OPTFLG_INTERNAL)
-
-struct BUILD_TABLE_INFO {
-  BUILD_TABLE_INFO();
-  ~BUILD_TABLE_INFO();
-
-  unsigned long remap_optflg;
-  int paramc;
-  int argc;
-  char *paramv[BUILD_TABLE_MAX_ARGS];
-  char *argv[BUILD_TABLE_MAX_ARGS];
-
-  bool ip_allow_check_enabled_p;
-  bool accept_check_p;
-  acl_filter_rule *rules_list; // all rules defined in config files as .define_filter foobar @src_ip=.....
-  UrlRewrite *rewrite;         // Pointer to the UrlRewrite object we are parsing for.
-
-  // Clear the argument vector.
-  void reset();
-
-  // noncopyable
-  BUILD_TABLE_INFO(const BUILD_TABLE_INFO &) = delete;            // disabled
-  BUILD_TABLE_INFO &operator=(const BUILD_TABLE_INFO &) = delete; // disabled
+class url_mapping;
+
+// remap rule types
+static constexpr ts::TextView REMAP_REGEX_PREFIX{"regex_"};
+static constexpr ts::TextView REMAP_MAP_TAG{"map"};
+static constexpr ts::TextView REMAP_REDIRECT_TAG{"redirect"};
+static constexpr ts::TextView REMAP_TEMPORARY_REDIRECT_TAG{"redirect_temporary"};
+static constexpr ts::TextView REMAP_REVERSE_MAP_TAG{"reverse_map"};
+static constexpr ts::TextView REMAP_WITH_REFERER_TAG{"map_with_referer"};
+static constexpr ts::TextView REMAP_WITH_RECV_PORT_TAG{"map_with_recv_port"};
```

Comment:
Why does this need ts::TextView? I know they are "compatible", but I thought we agreed that we should use std::string_view as much as possible, and use TextView when you need those additional features?

Either that, or we have to s/std::string_view/ts::TextView/g everywhere, I just find it really obnoxious that there's no rhyme or reason when to use one over the other.

Reply:
One reason is exemplified in #5053. For this PR, I tried to consistently use `TextView` because I need it in some places and you, as you do here, had complained about inconsistency. Therefore these are `TextView` because the working code needs `TextView` in the code that will use these, therefore done for the sake of consistency.